UI/src/view/linear_asset/laneModellingForm.js

```diff
@@ -0,0 +1,597 @@
+        if(_.isEmpty(even)){
+          nextLaneNumber = parseInt(odd.toString()[0] + '2');
+        }else{
+          nextLaneNumber = parseInt(_.max(even)) + 2;
+        }
+
+        selectedAsset.setNewLane(nextLaneNumber);
+        selectedAsset.setCurrentLane(nextLaneNumber);
+
+        defaultFormStructure.fields[0] = {label: 'Kaista', type: 'read_only_number', publicId: "lane_code", defaultValue: nextLaneNumber, weight: 6};
+        currentFormStructure = defaultFormStructure;
+
+        reloadForm($('#feature-attributes'));
+      }).prop("disabled", !_.isEmpty(even) && _.max(even).toString()[1] == 8));
+
+      var addRightLane = $('<li>').append($('<button class="btn btn-secondary">Lisää kaista oikealle puolelle</button>').click(function() {
+        $(".preview .lane").css('border', '1px solid white');
+
+        var nextLaneNumber = parseInt(_.max(odd)) + 2;
+
+        selectedAsset.setNewLane(nextLaneNumber);
+        selectedAsset.setCurrentLane(nextLaneNumber);
+
+        defaultFormStructure.fields[0] = {label: 'Kaista', type: 'read_only_number', publicId: "lane_code", defaultValue: nextLaneNumber, weight: 6};
+        currentFormStructure = defaultFormStructure;
+
+        reloadForm($('#feature-attributes'));
+      }).prop("disabled", !_.isEmpty(odd) && _.max(odd).toString()[1] == 9));
+
+      var selectedRoadLink = selectedAsset.getSelectedRoadlink();
+      var addByRoadAddress = isAddByRoadAddressActive ? $('<li>') : $('<li>').append($('<button class="btn btn-secondary">Lisää kaista tieosoitteen avulla</button>').click(function() {
+        isAddByRoadAddressActive = true;
+        selectedAsset.setInitialRoadFields();
+
+        reloadForm($('#feature-attributes'));
+      }).prop("disabled", _.isUndefined(selectedRoadLink.roadNumber) || _.isUndefined(selectedRoadLink.roadPartNumber) ||
+        _.isUndefined(selectedRoadLink.startAddrMValue) || selectedRoadLink.administrativeClass != 1 || selectedAsset.configurationIsCut() ||
+        !selectedAsset.haveNewLane() || _.filter(laneAssets, function (lane) {return lane.id !==0;}).length > 1));
+
+      return $('<ul class="list-lane-buttons">').append(addByRoadAddress).append(addLeftLane).append(addRightLane);
+    };
+
+    self.renderPreview = function(selectedAsset) {
+      var laneNumbers = _.map(selectedAsset.get(), function (lane){
+        return _.head(_.find(lane.properties, function (property) {
+          return property.publicId == "lane_code";
+        }).values).value;
+      });
+
+      return createPreviewHeaderElement(_.uniq(laneNumbers));
+    };
+
+    self.renderLaneButtons = function(selectedAsset) {
+      var isReadOnly = self._isReadOnly(selectedAsset);
+      var laneButtons = createLaneButtons(selectedAsset);
+      //Hide or show elements depending on the readonly mode
+      laneButtons.toggle(!isReadOnly);
+
+      return laneButtons;
+    };
+
+    self.renderForm = function (selectedAsset, isDisabled, isMassUpdate) {
```

Comment:
isMassUpdate its unused, check if it can be deleted

Reply:
Deleted